internal/mq/msgstream/mqwrapper/kafka/kafka_client.go

```diff
@@ -0,0 +1,103 @@
+package kafka
+
+import (
+	"strconv"
+
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/milvus-io/milvus/internal/log"
+	"github.com/milvus-io/milvus/internal/mq/msgstream/mqwrapper"
+	"go.uber.org/zap"
+)
+
+type kafkaClient struct {
+	basicConfig kafka.ConfigMap
+}
+
+func NewKafkaClientInstance(address string) *kafkaClient {
+	config := kafka.ConfigMap{
+		"bootstrap.servers":   address,
+		"socket.timeout.ms":   6000,
+		"api.version.request": true,
+	}
+
+	return &kafkaClient{basicConfig: config}
+}
+
+func cloneKafkaConfig(config kafka.ConfigMap) *kafka.ConfigMap {
+	newConfig := make(kafka.ConfigMap)
+	for k, v := range config {
+		newConfig[k] = v
+	}
+	return &newConfig
+}
+
+func (kc *kafkaClient) newConsumerConfig(group string, offset mqwrapper.SubscriptionInitialPosition) *kafka.ConfigMap {
+	newConf := cloneKafkaConfig(kc.basicConfig)
+
+	if offset == mqwrapper.SubscriptionPositionEarliest {
+		newConf.SetKey("auto.offset.reset", "earliest")
+	} else {
+		newConf.SetKey("auto.offset.reset", "latest")
+	}
+
+	newConf.SetKey("session.timeout.ms", 6000)
+	newConf.SetKey("group.id", group)
+	newConf.SetKey("enable.auto.commit", false)
+
+	//Kafka default will not create topics if consumer's the topics don't exist.
+	//In order to compatible with other MQ, we need to enable the following configuration,
+	//meanwhile, some implementation also try to consume a non-exist topic, such as dataCoordTimeTick.
+	newConf.SetKey("allow.auto.create.topics", true)
+	return newConf
+}
+
+func (kc *kafkaClient) CreateProducer(options mqwrapper.ProducerOptions) (mqwrapper.Producer, error) {
+	pp, err := kafka.NewProducer(&kc.basicConfig)
+	if err != nil {
+		log.Error("kafka create sync producer , error", zap.Error(err))
+		return nil, err
+	}
+
+	deliveryChan := make(chan kafka.Event, 1)
+	producer := &kafkaProducer{p: pp, deliveryChan: deliveryChan, topic: options.Topic}
+	return producer, nil
+}
+
+func (kc *kafkaClient) Subscribe(options mqwrapper.ConsumerOptions) (mqwrapper.Consumer, error) {
```

Comment:
why we don't do unscubstribe?

Reply:
Kafka consumer's Close API will unsubscribe